security/remote_devices/ssh-tunneling/local/device-ssh-proxy/docker-entrypoint.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/dumb-init /bin/sh
+
+# Copyright (C) 2020-2022 Intel Corporation
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+set -ex
+
+if [ "`stat -c '%u %g %a' /root/.ssh`" != "0 0 700" ]; then
+  chown 0:0 /root/.ssh
+  chmod 700 /root/.ssh
+fi
+if [ "`stat -c '%u %g %a' /root/.ssh/id_rsa`" != "0 0 600" ]; then
+  chown 0:0 /root/.ssh/id_rsa
+  chmod 600 /root/.ssh/id_rsa
+fi
+if [ "`stat -c '%u %g %a' /root/.ssh/id_rsa.pub`" != "0 0 600" ]; then
+  chown 0:0 /root/.ssh/id_rsa.pub
+  chmod 600 /root/.ssh/id_rsa.pub
+fi
+
+while true; do
+  scp -p \
+    -o StrictHostKeyChecking=no \
+    -o UserKnownHostsFile=/dev/null \
+    -P $TUNNEL_SSH_PORT \
+    /tmp/edgex/secrets/device-virtual/secrets-token.json $TUNNEL_HOST:/tmp/edgex/secrets/device-virtual/secrets-token.json
+  ssh \
+    -o StrictHostKeyChecking=no \
+    -o UserKnownHostsFile=/dev/null \
+    -p $TUNNEL_SSH_PORT \
+    $TUNNEL_HOST -- \
+    chown -Rh 2002:2001 /tmp/edgex/secrets/device-virtual
```

Comment:
? not sure i understand this chown here  with ssh command together

Reply:
The secrets-token.json gets copied over in the scp step is root user.  This step chown's it so that it can be read by device-virtual, which is a non-root user.

Reply:
ok